### PR TITLE
Duniverse v15.0.0

### PR DESCRIPTION
**Pull request by gridbugs**

### CODE REVIEW

File: opam [hunk 1]
```diff
@@ -20,33 +20,19 @@ maintainer: "Daniel Bünzli <daniel.buenzl i@erratique.ch>"
...
 bug-reports: "https://github.com/dbuenzli/uunf/issues"
 depends: [
+  "dune"
   "ocaml" {>= "4.03.0"}
-  "ocamlfind" {build}
-  "ocamlbuild" {build}
-  "topkg" {build & >= "1.0.3"}
-  "uucd" {dev & >= "15.0.0" & < "16.0.0"}
 ]
-depopts: ["uutf" "cmdliner"]
```
Comment (Leonidas-from-XIV) [inline opam:32]: Shouldn't `cmdliner` be moved to `depends`?

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,5 @@
 _b0
 _build
-src/uunf_data.ml
```
Comment (Leonidas-from-XIV) [inline .gitignore:3]: I wonder, is there maybe a way to generate this as part of a `dune` rule instead of running `pkg.ml`?

**Review by Leonidas-from-XIV — COMMENTED** [review]
Maybe there are some better ways to port this, that don't require calling `topkg` by the packager?

File: .gitignore (hunk 2 above)
Comment (gridbugs) [reply .gitignore:3]: That would certainly be possible. I chose to check in the result of running `pkg.ml` instead as that program downloads character data and then uses the data to generate some code and I didn't want the build target to depend on an internet connection if we could avoid it.

**Comment by gridbugs** [comment]
> Maybe there are some better ways to port this, that don't require calling `topkg` by the packager?

Do you mean checking in the code generated by running this package's build command? What problems are caused by doing this?

**Comment by Leonidas-from-XIV** [comment]
The main problem is that the dune port is "further away" from the source tarball. Which is not a problem per-se but usually the closer it is to the original source (e.g. no additional commands that need to be remembered to run) the easier it is to verify it is correct and the simpler it is to port it to the next version.

So sure, this is a valid way and we for sure don't want it to download random files from the internet at build time. I took a look at how this is generated upstream and it makes vaguely unhappy that it downloads and unpacks a random ZIP file, so the approach in this PR might indeed the best possible.

**Comment by gridbugs** [comment]
Would it be better to check in the raw xml file (the contents of the zip that gets downloaded) and have dune rules process the xml file into the generated ml files? That way we still avoid downloading a file as part of the build but the result is closer in some sense to the upstream source?

The problem with checking in the xml is that the process of porting the upstream package would still involve running the topkg command to download the file (or a custom curl/wget command I suppose), and if the porting process requires running a custom command and checking in the result anyway then I see no harm in running the custom command that generates the ml files instead (as we currently do) and that has the benefit of simplifying the dune files.

**Comment by emillon** [comment]
For things like that I would recommend checking in the downloaded source (it's probably in the release tarball no?). If you were upstream you could add a refresh rule with mode promote and alias `refresh`, so that `dune build @refresh` would redownload it, but I don't think we're going to update the data files independently of upstream releases.

File: opam (hunk 1 above)
Comment (gridbugs) [reply opam:32]: Why? the `Cmdliner` module is only referred to in tests.

**Comment by gridbugs** [comment]
@Leonidas-from-XIV does anything more need to happen here or is this good to merge?

**Review by Leonidas-from-XIV — APPROVED** [review]

**Pull request merged**